src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -206,14 +208,26 @@ public PrivilegesEvaluatorResponse evaluate(final User user, String action0, fin
             action0 = PutMappingAction.NAME;
         }
 
+        final PrivilegesEvaluatorResponse presponse = new PrivilegesEvaluatorResponse();
+
         final TransportAddress caller = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
-        final Set<String> mappedRoles = (injectedRoles == null) ? mapRoles(user, caller) : injectedRoles;
+        Set<String> mappedRoles = (injectedRoles == null) ? mapRoles(user, caller) : injectedRoles;
+        final String assumeRolesString = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_ASSUME_ROLES);
+        if(assumeRolesString != null) {
+            HashSet<String> assumeRoleSet = new HashSet<>(Arrays.asList(assumeRolesString.split(",")));
+            if(!mappedRoles.containsAll(assumeRoleSet)) {
```

Comment:
I don't follow that logic. What is the significance of comparing role names? What if I have two roles with the same name but with different permissions? What if two roles have different names but the same permissions? Please open new issue and describe use case and how that use case will be solved.

Reply:
Added here: https://github.com/opensearch-project/security/issues/1337

Reply:
What this check enforce and why is it needed?

Reply:
The mappedRoles contains all the roles associated with a particular user. The `assumeRoleSet ` transients populated by plugins can force the evaluation to restrict on `assumeRolesSet`. With this:
 - The user is confirmed to be associated with these roles(assumeRolesSet) and
 -  If the permissions evaluation succeeds, the assume roles have the required permissions to invoke transport actions

If either assumeRolesSet(from the transients) is not associated with user or if there are these roles associated with the user but doesn't have permissions to invoke transport actions, then security exception is thrown.

Reply:
I see, so this change is not directly related to the role injection? If yes, please split PR into two or more (if there are more independent changes) separate PRs.

Reply:
Done: https://github.com/opensearch-project/security/pull/1367